app/controllers/audits_controller.rb

```diff
@@ -12,6 +12,13 @@ def show
 
   def save
     audit.user = current_user
+    # Get the next_item before updating the audit. As the
```

Comment:
> Can you please add a test to drive this fix?

Yep, will see if I can do that today.

> Can you think of a way to get rid of the comment via:
> 
>     Writing more expressive code

I had a think about this yesterday, mostly while writing https://github.com/alphagov/content-performance-manager/pull/223 , and I haven't come up with anything yet.

>     Describing what you try to fix in the test?
> 
> Regardless the fix, it feels to me that it assumes that we rely on having the next item calculation precalculated, but as you described in #198 (comment), it has many drawbacks.
> 
> So, it might be better to find a good solution for navigation rather than adding quick fixes?

In this case, which is the case where the Save button is pressed, the behaviour is simpler than the Next link, as the decision about the next item is deferred until the point at which the user clicks save (rather than the Next link, where its currently decided when the page is rendered). My current thinking on this is that its independent of the navigation, as its just about making sure that the next item can be found when saving an audit. Does this make sense?